bootstrap.sh

```diff
@@ -131,7 +131,35 @@ echo "Exit now (ctrl-c) if this is incorrect"
 echo
 
 sleep 5
+
+# Wait up to 30 seconds for the package lists lock to become available.
+lock_wait_start=$(date +%s)
+printed_status=false
+while fuser /var/lib/apt/lists/lock; do
+    if [ "${printed_status}" = false ]; then
+        echo "Waiting for another process to update package lists"
+        printed_status=true
+    fi
+    current_time=$(date +%s)
+    elapsed_time=$((current_time - lock_wait_start))
+    if [ "${elapsed_time}" -gt 30 ]; then break; fi
```

Comment:
So we leave it to `apt-get update` to fail if lock is closed?

Reply:
Yes, we could also print a warning here, but I think the error from the package manager is clear enough. The wait is mostly for us when we are building the image. A human operator can look into what is holding the lock and rerun the build.